lib/Dialect/D2M/Transforms/GridSelection.cpp

```diff
@@ -759,10 +760,85 @@ static bool hasTTNNOperands(d2m::GenericOp genericOp) {
     if (operand.getDefiningOp<ttir::TTNNMetalLayoutCastOp>()) {
       return true;
     }
+    // Bypass any views applied in TTIRToD2M to check if viewed tensor itself is
+    // the result of a TTNNMetalLayoutCastOp.
+    else if (auto view = operand.getDefiningOp<d2m::ViewLayoutOp>()) {
```

Comment:
:warning: **llvm\-else\-after\-return** :warning:
do not use `` else `` after `` return ``
```suggestion
    if (auto view = operand.getDefiningOp<d2m::ViewLayoutOp>()) {
```

Reply:
fixed in latest version 👍